### PR TITLE
Review branch for ISC UDS - Microsoft Sentinel connector

### PR DESCRIPTION
**Pull request by annish-py**
Implemented Translate, Transmit and Translate-Results for Microsoft Azure Sentinel data provider. Please find the README.md file for brief details on this connector.

### CODE REVIEW

File: stix_shifter/stix_translation/src/modules/azure_sentinel/README.md [hunk 1]
```diff
@@ -0,0 +1,375 @@
+# Azure Sentinel - ISC UDS Connector
```
Comment (delliott90) [inline stix_shifter/stix_translation/src/modules/azure_sentinel/README.md]: Remove reference to ISC UDS.

File: stix_shifter/stix_translation/src/modules/azure_sentinel/json/from_stix_map.json [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+      "value": ["hostStates.fqdn", "hostStates.netBiosName"]
+    }
+  },
+  "user-account": {
+    "fields": {
+      "user_id": ["userStates.accountName"],
+      "account_last_login": ["userStates.logonDateTime"]
+    }
+  },
+  "x_com_msazure_sentinel": {
```
Comment (delliott90) [inline stix_shifter/stix_translation/src/modules/azure_sentinel/json/from_stix_map.json]: Use dashes for the custom STIX objects and underscores for the object properties. So in this case it should be `x-com-msazure-sentinel`.

File: stix_shifter/stix_translation/src/modules/azure_sentinel/README.md (hunk 1 above)
Comment (amaledwin) [reply stix_shifter/stix_translation/src/modules/azure_sentinel/README.md]: Removed ISC UDS reference from README.md

File: stix_shifter/stix_translation/src/modules/azure_sentinel/json/from_stix_map.json (hunk 2 above)
Comment (amaledwin) [reply stix_shifter/stix_translation/src/modules/azure_sentinel/json/from_stix_map.json]: underscore('_') is replaced with hyphen('-')  in custom attributes.

File: stix_shifter/stix_translation/src/modules/azure_sentinel/query_constructor.py [hunk 3]
```diff
@@ -0,0 +1,406 @@
...
+                    raise SearchFeatureNotSupportedError('"{operator}" operator is not supported for '
+                                                         '"{stix_field}" attribute'
+                                                         .format(operator=expression.comparator.name.upper(),
+                                                                 stix_field=stix_field))
+                value = self._format_value_without_quotes(value)
+
+            if stix_field not in ['provider', 'vendor']:
+                value = self._format_value_to_lower_case(value)
+
+            self.COUNTER += 1
```
Comment (mdazam1942) [inline stix_shifter/stix_translation/src/modules/azure_sentinel/query_constructor.py]: some comments would be helpful for this COUNTER global variable

**Comment by delliott90** [comment]
I tried to pull down some new changes made to the master branch and noticed your repo doesn't have the latest changes to the master branch. Probably not a huge deal but it wouldn't hurt to fetch the latest upstream branch to your repo.

File: stix_shifter/stix_translation/src/modules/azure_sentinel/query_constructor.py (hunk 3 above)
Comment (annish-py) [reply stix_shifter/stix_translation/src/modules/azure_sentinel/query_constructor.py]: Comment line updated for the COUNTER variable. COUNTER is used to form sequential lambda function names for OData4 queries per comparison observation.
e.g. processes/any(query1:contains(tolower(query1/path), 'c:\\windows\\system32')) and processes/any(query2:contains(tolower(query2/name), 'exe')) '''. Here 'query1' and 'query2' are lambda function names used in the OData queries for better readability and query execution.

**Comment by annish-py** [comment]
> I tried to pull down some new changes made to the master branch and noticed your repo doesn't have the latest changes to the master branch. Probably not a huge deal but it wouldn't hurt to fetch the latest upstream branch to your repo.

**Comment by annish-py** [comment]
> > I tried to pull down some new changes made to the master branch and noticed your repo doesn't have the latest changes to the master branch. Probably not a huge deal but it wouldn't hurt to fetch the latest upstream branch to your repo.

**Comment by mdazam1942** [comment]
@annish-py please resolve the conflicts.

File: stix_shifter/stix_transmission/src/modules/azure_sentinel/azure_sentinel_api_client.py [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+    """API Client to handle all calls."""
+    def __init__(self, connection, configuration):
+        """Initialization.
+        :param connection: dict, connection dict
+        :param configuration: dict,config dict"""
+
+        headers = dict()
+        url_modifier_function = None
+        auth = configuration.get('auth')
+        self.endpoint = 'beta/security/alerts'
```
Comment (mdazam1942) [inline stix_shifter/stix_transmission/src/modules/azure_sentinel/azure_sentinel_api_client.py]: just wondering, do you know when they are removing the `beta/ ` from the endpoint path?

Comment (annish-py) [reply stix_shifter/stix_transmission/src/modules/azure_sentinel/azure_sentinel_api_client.py]: @mdazam1942 The current code is now modified to take an optional configuration parameter(api_version), which could be either 'beta' or 'v1.0'. By default, if the parameter is not supplied the api_version is set to 'v1.0' for building the sentinel API endpoint.

**Comment by annish-py** [comment]
> @annish-py please resolve the conflicts.



> @annish-py please resolve the conflicts.

@mdazam1942 Fixed conflicts and updated the code to latest.

File: stix_shifter/stix_transmission/src/modules/azure_sentinel/azure_sentinel_api_client.py (hunk 4 above)
Comment (mdazam1942) [reply stix_shifter/stix_transmission/src/modules/azure_sentinel/azure_sentinel_api_client.py]: this is better.i replied to your email and asked few question about how the user would get the api version. If the user know the version then we can ask them to put in the connection object.

I would actually put that inside `connection` object in the options attribute. like this:

`api_version = connection.get('options', {}).get('api_version', default_api_version)`

Comment (annish-py) [reply stix_shifter/stix_transmission/src/modules/azure_sentinel/azure_sentinel_api_client.py]: The api-version information for Azure sentinel alerts endpoint is not available in the console, we can obtain that from Microsoft public technical documents only. As preferred, we have reverted the connector code to utilize the hard-coded api- version(‘v1.0’).

Comment (mdazam1942) [reply stix_shifter/stix_transmission/src/modules/azure_sentinel/azure_sentinel_api_client.py]: sounds good

**Pull request merged**